weave/trace_server/trace_server_interface.py

```diff
@@ -212,6 +221,26 @@ class TableSchemaForInsert(BaseModel):
     rows: list[dict[str, Any]]
 
 
+class OtelExportReq(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    project_id: str
+    traces: ExportTraceServiceRequest
+    wb_user_id: Optional[str] = Field(None, description=WB_USER_ID_DESCRIPTION)
+
+
+# Spec requires that the response be of type Export<signal>ServiceResponse
+# https://opentelemetry.io/docs/specs/otlp/
+class OtelExportRes(BaseModel):
```

Comment:
I don't quite understand why you still went with BaseModel here instead of just a simple class given the workaround needed. What do we get from BaseModel here?

Reply:
Anything which expects a Basemodel such as the generic endpoint caller in the http server can take this as a parameter. We only really care about the nested field and validation is skipped for that so it adds a bit of flexibility at no cost.